aspnetcore/web-api/jsonpatch.md

```diff
@@ -159,81 +95,211 @@ The preceding action method example calls an overload of `ApplyTo` that takes mo
+patchDoc!.ApplyTo(person, jsonPatchError =>
+    {
+        errors ??= new ();
+        var key = jsonPatchError.AffectedObject.GetType().Name;
+        if (!errors.ContainsKey(key))
+        {
+            errors.Add(key, new string[] { });
+        }
+        errors[key] = errors[key].Append(jsonPatchError.ErrorMessage).ToArray();
+    });
+if (errors != null)
+{
+    // Print the errors
+    foreach (var error in errors)
+    {
+        Console.WriteLine($"Error in {error.Key}: {string.Join(", ", error.Value)}");
+    }
+}
 
-:::code language="json" source="~/web-api/jsonpatch/snippets/replace.json":::
+// Output updated object
+Console.WriteLine(JsonSerializer.Serialize(person, serializerOptions));
+```
 
-## The move operation
+The previous example results in the following output:
 
-* If `path` points to an array element: copies `from` element to location of `path` element, then runs a `remove` operation on the `from` element.
-* If `path` points to a property: copies value of `from` property to `path` property, then runs a `remove` operation on the `from` property.
-* If `path` points to a nonexistent property:
-  * If the resource to patch is a static object: the request fails.
-  * If the resource to patch is a dynamic object: copies `from` property to location indicated by `path`, then runs a `remove` operation on the `from` property.
+```output
+Error in Person: The current value 'John' at path 'FirstName' is not equal 
+to the test value 'Jane'.
+{
+    "firstName": "John",
+    "lastName": "Smith",              <<< Modified!
+    "email": "janedoe@gmail.com",     <<< Modified!
+    "phoneNumbers": []
+}
+```
 
-The following sample patch document:
+## Mitigating security risks
 
-* Copies the value of `Orders[0].OrderName` to `CustomerName`.
-* Sets `Orders[0].OrderName` to null.
-* Moves `Orders[1]` to before `Orders[0]`.
+When using the `Microsoft.AspNetCore.JsonPatch.SystemTextJson` package, it's critical to understand and mitigate potential security risks. The following sections outline the identified security risks associated with JSON Patch and provide recommended mitigations to ensure secure usage of the package.
 
-:::code language="json" source="~/web-api/jsonpatch/snippets/move.json":::
+> [!IMPORTANT]
+> ***This is not an exhaustive list of threats.*** App developers must conduct their own threat model reviews to determine an app-specific comprehensive list and come up with appropriate mitigations as needed. For example, apps which expose collections to patch operations should consider the potential for algorithmic complexity attacks if those operations insert or remove elements at the beginning of the collection.
 
-## The copy operation
+By running comprehensive threat models for their own apps and addressing identified threats while following the recommended mitigations below, consumers of these packages can integrate JSON Patch functionality into their apps while minimizing security risks.
```

Comment:
Very long sentence, And most of it seems to be repeated in the following text

Reply:
Resolved.  Yes, that really needed to be whittled down!